hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/traceability/MirrorOperationTracer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.web3.evm.contracts.execution.traceability;
+
+import static com.hedera.mirror.web3.evm.utils.EvmTokenUtils.entityIdNumFromEvmAddress;
+
+import com.hedera.mirror.web3.evm.properties.TracingProperties;
+import com.hedera.node.app.service.evm.contracts.execution.traceability.HederaEvmOperationTracer;
+import java.nio.charset.StandardCharsets;
+import javax.inject.Named;
+import lombok.CustomLog;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.operation.Operation;
+
+@CustomLog
+@Named
+@RequiredArgsConstructor
+public class MirrorOperationTracer implements HederaEvmOperationTracer {
+
+    private final TracingProperties tracingProperties;
+
+    @Override
+    public void tracePostExecution(final MessageFrame currentFrame, final Operation.OperationResult operationResult) {
+        if (tracingProperties.isEnabled()) {
```

Comment:
Fail fast to reduce cognitive complexity and nested conditionals.

```suggestion
        if (!tracingProperties.isEnabled()) {
            return;
        }
```